### PR TITLE
Change configuration loading to cosmiconfig

### PR DESCRIPTION
**Pull request by bunysae**
Changed configuration loading to cosmiconfig module. As @sindresorhus [mention](https://github.com/xojs/xo/pull/279#issuecomment-407250935) YAML is disabled.
Fixes issue #330 and superseded PR #331 as this seems no longer maintained.

### CODE REVIEW

File: lib/options-manager.js [hunk 1]
```diff
@@ -155,14 +156,32 @@ const normalizeOptions = options => {
...
+	const moduleName = 'xo';
+	// Yaml is disabled
+	const searchPlaces = ['package.json',
+		`.${moduleName}rc`,
+		`.${moduleName}rc.json`,
+		`.${moduleName}rc.js`,
+		`${moduleName}.config.js`];
+	const loaders = {noExt: defaultLoaders['.json']};
+	// Change working directory for load config
+	process.chdir(options.cwd);
```
Comment (pvdlg) [inline lib/options-manager.js]: Instead of changing the working directory directory, pass `options.cwd` to comisconfig in option `searchFrom`

**Comment by sindresorhus** [comment]
Thanks for working on this. It would be good with some more tests. And it also needs to be documented in the readme.

File: lib/options-manager.js (hunk 1 above)
Comment (bunysae) [reply lib/options-manager.js]: Done. I didn't saw this option before.

**Comment by bunysae** [comment]
I added tests for every variety and changed the docs with my latest commit.

**Comment by sindresorhus** [comment]
Sorry @bunysae... I went with https://github.com/xojs/xo/pull/425 as it was much more complete.

**Pull request closed without merging**